### DISCUSSION TITLE
HC05 or JDY-23 bluetooth module connection for esp32lolin

### DISCUSSION THREAD

**Discussion by paulsteigel** [syssi/esphome-jk-bms#491] [Q&A] [answered]
Hi @syssi ,
I am using a Esp32 lolin without bluetooth. I am thinking about using this board with JDY-23 module (similar to HC05) for testing bluetooth connection. 
Do you have any guidance on how to do this. I am trying to set up an uart connection to the module but can not find a way to have this uart work with ble_client using your code!
Many thanks,

**Comment by syssi** [comment]
IMO the mentioned bluetooth modules doesn't support bluetooth low energy. You should invest the money to buy a ESP32 with bluetooth support instead.

**Reply by paulsteigel** [reply]
Thanks @syssi,
The reason why I am asking this question is not about investment at all. What I am wondering is can it be possible to do that. However, you have explained already then!
Many thanks again for your response!
Best

**Reply by syssi** [reply, marked as answer]
If you have a separate bluetooth module capable of bluetooth low energy you could implement a `ble_client` component implementation ("the driver") to talk to the module. IMO BLE is pretty complex and this will be a huge task. My implementation just uses some BLE methods of the ESP-IDF implementation / BLE stack of the ESP32.

**Reply by paulsteigel** [reply]
Many thanks, I will try a bit to see how I can learn from your approach to make it work on the JDY-23 (in fact similar to Hc05) and they do support low energy power (I check the datasheet)
